pkg/datasources/roles.go

```diff
@@ -0,0 +1,80 @@
+package datasources
+
+import (
+	"database/sql"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+)
+
+var rolesSchema = map[string]*schema.Schema{
+	"roles": {
```

Comment:
in addtion to roles, we also need to include an optional argument for "pattern" https://docs.snowflake.com/en/sql-reference/sql/show-roles.html